principles.rst

```diff
@@ -303,3 +309,126 @@ down GHC even when those features are not active. Yet this principle is importan
 as we hope not to make the current situation worse.
 
 From `#378`_, slightly generalized.
+
+
+
+GHC stability principles
+--------------------------
+
+The Haskell ecosystem has a built-in tension between stability and innovation.
+
+One the one hand, breaking changes impose heavy costs:
+
+* Users keep having to update their code
+* Library authors keep having to update their code
+* Updating to a later version of a library can in turn force taking on new dependencies.
+
+These difficulties add friction, discourage adoption, and make Haskell an unpredictable investment.
+
+On the other hand, GHC has always sought to be a laboratory for innovation.  If the stability guarantees are too onerous, we risk imposing a different set of costs:
+
+* Volunteers may get discouraged by the hoops they have to jump through to get a change agreed.
+* The language may become stuck in a local optimum, because moving to a better design would require breaking changes.
+* GHC may become encrusted with cruft and workarounds that exist to preserve existing, but undesirable, behaviour.
+
+We can't navigate this tension with simple all-or-nothing rules.  We have to take each case on its merits, aware of both sets of costs.  Rather than starting the debate from scratch each time, we can have some general rules, or principles, to guide us.  The `GHC base library proposal <https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst>`_ made a start in this direction; this section discusses GHC itself.
+
+Assumptions
+~~~~~~~~~~~~
+
+We assume:
+
+* That we have adopted the  `GHC base library proposal <https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst>`_, which establishes the ``ghc-experimental`` package.
+* That we have identified some extensions as "Experimental".  These extensions may well change, and are subject to
+  much weaker stability goals.  `GHC Proposal #601 <https://github.com/ghc-proposals/ghc-proposals/pull/601>`_, describes a plan along these lines, but all we need for this document is a list of Experimental extensions.
+
+Terminology
+~~~~~~~~~~~~~~~
+
+We define a **stable Haskell package** as follows. A stable Haskell package
+
+* Does not use Experimental extensions.
+* Does not use experimental features.  (Examples: use of the Javascript or Wasm back end, builds on non-tier-1 platforms.)
+* Does not rely on explicitly-undefined behaviour. (Example: ``INCOHERENT`` instance selection.)
+* Does not use ``-Werror`` in its default build configuration.
+* Explicitly specifies a language edition (``Haskell98``, ``GHC2021``), in the source code or the build configuration.
+* Depends only on stable Haskell packages.
```

Comment:
Sorry to be a curmudgeon about this again. But the proportion of packages that don't depend transitively on Template Haskell (which is, rightly, considered unstabled) is tiny. If we state stability goals only for packages which don't depend transitively on Template Haskell, then we're making a largely theoretical statement.

Proposal: drop this condition from the definition of stable package, and GR1 becomes “if the dependencies work, then the package work”. Much more compelling goal in my opinion.